scripts/tasks/src/type-check.ts

```diff
@@ -0,0 +1,43 @@
+import * as fs from 'fs';
+
+import { logger } from 'just-scripts';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { exec } from 'just-scripts-utils';
+
+import { getTsPathAliasesConfig } from './utils';
+
+export function typeCheck() {
+  const { isUsingTsSolutionConfigs, tsConfigFileContents, tsConfigs, tsConfigFilePaths } = getTsPathAliasesConfig();
+
+  if (!isUsingTsSolutionConfigs) {
+    logger.info('project is not using TS solution config. skipping...');
+    return;
+  }
+
+  const content = tsConfigFileContents.root;
+  const config = tsConfigs.root;
+  const configPath = tsConfigFilePaths.root;
+
+  if (!(content && config)) {
+    return;
+  }
+
+  // turn off path aliases.
+  // @ts-expect-error - bad just-scripts ts types
+  config.compilerOptions.paths = {};
+  fs.writeFileSync(configPath, JSON.stringify(config, null, 2), 'utf-8');
```

Comment:
there is no other way how to turn off path aliases in tsc `build` mode. 

We could re-create configs on the fly, remap paths etc, but that would be additional processing and logic that would need to be maintained and also would slow down the task in general, thus this solution feels the fastest/most easy to maintain/understand.